src/api/public/apidocs/paths/source_project_name_package_name_meta.yaml

```diff
@@ -6,6 +6,12 @@ get:
   parameters:
     - $ref: '../components/parameters/project_name.yaml'
     - $ref: '../components/parameters/package_name.yaml'
+    - in: query
+      name: deleted
+      schema:
+        type: string
+      description: Set to `1` to retrieve the package metadata of a deleted package.
+      example: 1
```

Comment:
no, it is a boolean, just the existens is enough

Reply:
Checking locally with `&deleted=a` returns a `not boolean` error:
```
> curl -i -u Admin:XXXXXXXX -H 'Content-Type: application/xml; charset=utf-8' -X GET "http://localhost:3000/source/home:Admin/hey/_meta?deleted=a"
<status code="400">
  <summary>not boolean</summary>
</status>
>
```
The backend also defines the `deleted` parameter as bool: https://github.com/openSUSE/open-build-service/blob/d5f0c54b8ad3163115cf8c986302b482e4934dc2/src/backend/bs_srcserver#L7675

Reply:
I tested locally with a deleted `hey` package.

These are some other tests:

- With `deleted=1`:
   ```
   > curl -i -u Admin:XXXXXXXX -H 'Content-Type: application/xml; charset=utf-8' -X GET "http://localhost:3000/source/home:Admin/hey/_meta?deleted=1"
   <package name="hey" project="home:Admin">
     <title></title>
     <description></description>
   </package>
   ```
- With `deleted=0`:
   ```
   > curl -i -u Admin:XXXXXXXX -H 'Content-Type: application/xml; charset=utf-8' -X GET "http://localhost:3000/source/home:Admin/hey/_meta?deleted=0"
   <status code="404">
     <summary>package 'hey' does not exist in project 'home:Admin'</summary>
     <details>404 package 'hey' does not exist in project 'home:Admin'</details>
   </status>
   ```
- Without the `deleted` parameter:
   ```
   > curl -i -u Admin:XXXXXXXX -H 'Content-Type: application/xml; charset=utf-8' -X GET "http://localhost:3000/source/home:Admin/hey/_meta"
   <status code="unknown_package">
     <summary>Package not found: home:Admin/hey</summary>
   </status>
   ```

Reply:
And lastly, if we are talking about `type:` being `string` and not `boolean`, I chose `string` because the `boolean` type is only for representing the values `true` and `false`.

See the reference documentation: https://swagger.io/docs/specification/data-models/data-types/#boolean

Reply:
You can see in the review-app how it will look like, [here](https://obs-reviewlab.opensuse.org/adrianschroeter-fix_deleted_meta/apidocs/index.html#/Sources%20-%20Packages/get_source__project_name___package_name___meta).

Reply:
http://localhost:3000/source/home:Admin/hey/_meta?deleted

is supposed to be used and should work.

Reply:
deleted=0 is not really defined what will happen though, please do not mention this anywhere.

(yes, some need it as hacks to workaround bugs in some cases, but really nothing what should be documented)

Reply:
> http://localhost:3000/source/home:Admin/hey/_meta?deleted
> 
> is supposed to be used and should work.

Confirmed. - With `deleted`:
   ```
   > curl -i -u Admin:XXXXXXXX -H 'Content-Type: application/xml; charset=utf-8' -X GET "http://localhost:3000/source/home:Admin/hey/_meta?deleted"
   <package name="hey" project="home:Admin">
     <title></title>
     <description></description>
   </package>
   ```

Reply:
> deleted=0 is not really defined what will happen though, please do not mention this anywhere.

I removed the `0` option from the `enum`. Now it is not mentioned.

> (yes, some need it as hacks to workaround bugs in some cases, but really nothing what should be documented)

I prefer to document all the endpoints with all their parameters in the reference documentation. We can always decide what to include in the long form API documentation, yet to be written.

I have modified the documentation commit I added to adapt it to the requirements of this boolean `deleted` parameter. Please change the commit as you wish, or even delete it if you think it is better that way.